stepfunctions/src/main/scala/awscala/stepfunctions/StateMachine.scala

```diff
@@ -0,0 +1,46 @@
+package awscala.stepfunctions
+
+import awscala.DateTime
+import awscala.stepfunctions.ArnFormat.ResourceArn
+import com.amazonaws.services.stepfunctions.model._
+
+import scala.collection.JavaConverters._
+
+case class StateMachine(arn: String) {
+  def name: String = ArnFormat.parseArn(arn, ResourceArn)
+
+  def startExecution(input: String, name: String = null)(implicit steps: StepFunctions): Execution = {
+    val exec =
+      steps.startExecution {
+        new StartExecutionRequest()
+          .withStateMachineArn(arn)
+          .withInput(input)
+          .withName(name)
+      }
+    Execution(exec.getExecutionArn, new DateTime(exec.getStartDate))
+  }
+
+  def execution(name: String)(implicit steps: StepFunctions): Option[Execution] = executions().find(_.name == name)
+
+  def executions()(implicit steps: StepFunctions): Seq[Execution] =
+    steps
+      .listExecutions(new ListExecutionsRequest().withStateMachineArn(arn))
+      .getExecutions
```

Comment:
Can this method return null value? If it can return null, we need to care the case.

Reply:
Nope, this and the other `list` ones will return an empty list not null. But this response is paged so I should make this use a `Sequencer` like the others.